lab_1/task_2/update-pipeline-definition.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+function help() {
+    echo "Usage: update-pipeline-definition.sh <pipeline definition file> [--configuration <configuration>] [--owner <owner>] [--branch <branch>] [--poll-for-source-changes <true/false>] [--help]"
+    echo ""
+    echo "Options:"
+    echo "  --configuration             Pipeline configuration, e.g. development, staging, production."
+    echo "  --owner                     The GitHub owner/account name of repository."
+    echo "  --branch                    Branch to use for the pipeline."
+    echo "  --repo                      The GitHub repo that will be used by pipeline."
+    echo "  --poll-for-source-changes   Enable or disable polling for source code changes."
+    echo "  --help                      Show this help message."
+    exit 1
+}
+
+if [ $1 == '--help'  ]; then
+    help
+fi
+
+if ! [ -x "$(command -v jq)" ]; then
+  echo 'Error: jq is not installed.' >&2
+  echo 'Please install jq before running this script. Visit https://stedolan.github.io/jq/download/ for installation instructions.' >&2
+  exit 1
+fi
+
+
+if [[ ! -f "$1" || $(file --mime-type -b "$1") != "application/json" ]]; then
+    echo "Error: First argument is not a valid JSON file."
+    exit 1
+fi
+
+INITIAL_NUM_OF_ARGS=$#
+PIPELINE_DEFINITION_FILE=$1
+shift
+
+if ! jq -e '.pipeline.version' "$PIPELINE_DEFINITION_FILE" >/dev/null; then
+    echo "Error: Version property is missing. Please update pipeline definition file."
+    exit 1
+fi
+
+if ! jq '.pipeline.stages[0].actions[0].configuration.Branch' $PIPELINE_DEFINITION_FILE >/dev/null; then
+    echo "Error: Branch property is missing. Please update pipeline definition file."
+    exit 1
+fi
+
+if ! jq '.pipeline.stages[0].actions[0].configuration.Owner' $PIPELINE_DEFINITION_FILE >/dev/null; then
+    echo "Error: Owner property is missing. Please update pipeline definition file."
+    exit 1
+fi
+
+if ! jq '.pipeline.stages[0].actions[0].configuration.Repo' $PIPELINE_DEFINITION_FILE >/dev/null; then
+    echo "Error: Repo property is missing. Please update pipeline definition file."
+    exit 1
+fi
+
+if ! jq '.pipeline.stages[0].actions[0].configuration.PollForSourceChanges' $PIPELINE_DEFINITION_FILE >/dev/null; then
+    echo "Error: PollForSourceChanges property is missing. Please update pipeline definition file."
+    exit 1
+fi
+
+if ! jq '.pipeline.stages[1].actions[].configuration.EnvironmentVariables' $PIPELINE_DEFINITION_FILE >/dev/null; then
```

Comment:
there is still fixed stage's action configuration used. I assume that you just missed this for fix